### PR TITLE
#78: Added new PhpUnit extension and event subscriber to support PhpUnit v10 and deleted redundant PushDocumentation artisan command

### PR DESCRIPTION
**Pull request by vitgrams**
…

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by vitgrams** [comment]
@DenTray Implemented for phpUnit v 10. These changes will break backward compatibility.

**Review by NikitaYakovlev — APPROVED** [review]

File: src/Tests/PhpUnitEventSubscribers/SwaggerSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace RonasIT\Support\AutoDoc\Tests\PhpUnitEventSubscribers;
+
+use Illuminate\Contracts\Console\Kernel;
+use Illuminate\Foundation\Application;
+use PHPUnit\Event;
+use PHPUnit\Event\Test\AfterLastTestMethodFinished;
+use RonasIT\Support\AutoDoc\Services\SwaggerService;
+
+final class SwaggerSubscriber implements Event\Test\AfterLastTestMethodFinishedSubscriber
```
Comment (DenTray) [inline src/Tests/PhpUnitEventSubscribers/SwaggerSubscriber.php:11]:
```suggestion
final class SwaggerSubscriber implements AfterLastTestMethodFinishedSubscriber
```

Comment (DenTray) [inline src/Tests/PhpUnitEventSubscribers/SwaggerSubscriber.php:11]:
```suggestion
final class SwaggerSaveDocumentationSubscriber implements Event\Test\AfterLastTestMethodFinishedSubscriber
```

**Review by DenTray — CHANGES_REQUESTED** [review]

**Comment by vitgrams** [comment]
@DenTray Resolved review conversations.

**Review by DenTray — APPROVED** [review]

**Comment by DenTray** [comment]
@vitgrams could you please cover new code by test cases

**Comment by vitgrams** [comment]
@DenTray Increased code coverage by unit tests. I can't write test method for PhpUnit extension class because `notify()` method has mandatory event argument which is the object of the final class.

**Comment by vitgrams** [comment]
@DenTray Please note that the changes from 78_save_temp_doc_in_file_for_local_and_storage_drivers branch have been merged into current branch because `rebase` operation is not allowed here.

**Comment by DenTray** [comment]
@vitgrams could you please resolve conflicts

**Comment by DenTray** [comment]
required new major version, waiting for stable master

**Comment by DenTray** [comment]
@Goodmain could you please resolve conglicts

**Comment by Goodmain** [comment]
@DenTray Done

**Review by DenTray — APPROVED** [review]

**Pull request merged**